src/window.rs

```diff
@@ -1516,3 +1527,21 @@ impl Default for WindowLevel {
         Self::Normal
     }
 }
+
+/// IME purposes.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[non_exhaustive]
+pub enum ImePurpose {
```

Comment:
I think we should also document which platform supports which variant exactly.

Reply:
Currently, only Wayland is implemented, so there are only variants that work there. If other platforms are implemented, this should be documented per variant.

Reply:
I guess it's fine then, but you should have a section like we have for everything else.